prov/coll/src/coll.h

```diff
@@ -0,0 +1,198 @@
+
+/*
+ * Copyright (c) 2022 Intel Corporation, Inc.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _COLL_H_
+#define _COLL_H_
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include <rdma/fabric.h>
+#include <rdma/fi_errno.h>
+#include <rdma/fi_cm.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_eq.h>
+
+#include <ofi.h>
+#include <ofi_enosys.h>
+#include <ofi_util.h>
+#include <ofi_list.h>
+#include <ofi_lock.h>
+#include <ofi_proto.h>
+#include <ofi_iov.h>
+#include <ofi_hmem.h>
+#include <ofi_prov.h>
+#include <ofi_atomic.h>
+
+#define COLL_IOV_LIMIT 4
+#define COLL_MR_MODES	(OFI_MR_BASIC_MAP | FI_MR_LOCAL)
+#define COLL_TX_OP_FLAGS (0)
+#define COLL_RX_OP_FLAGS (0)
+
+enum {
+	COLL_RX_SIZE = 65536,
+	COLL_TX_SIZE = 16384,
+};
+
+struct coll_fabric {
+	struct util_fabric util_fabric;
+};
```

Comment:
Any reason to wrap util_fabric?

Reply:
Not really. I started with a skeleton, expecting to add fields after that, but ended up with none. Can change to use util_fabric directly.